### PR TITLE
Support paging of the application owner

### PR DESCRIPTION
**Pull request by ajordens**
image

image

### CODE REVIEW

**Comment by ajordens** [comment]
PTAL @anotherchrisberry.

Not particularly sold on the location it's launched from but figure that's easily changed.

File: app/scripts/modules/core/application/modal/pageApplicationOwner.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  <task-monitor monitor="taskMonitor"></task-monitor>
+  <modal-close></modal-close>
+  <div class="modal-header">
+    <h3 data-purpose="modal-header">Page {{application.name}} Owner</h3>
+  </div>
+
+  <form role="form" class="container-fluid" novalidate name="pageApplicationOwnerForm">
+    <div class="modal-body">
+
+      <div class="row" style="margin-top:10px; margin-bottom:10px">
```
Comment (ajordens) [inline app/scripts/modules/core/application/modal/pageApplicationOwner.html]: This is effectively the same as the `reason` directive safe for different placeholder text.

**Comment by anotherchrisberry** [comment]
How would you feel about moving the button to the config tab:

screen shot 2016-06-21 at 10 17 45 pm

(with tooltip)
screen shot 2016-06-21 at 10 18 00 pm

(or with text instead of tooltip)
screen shot 2016-06-21 at 10 20 05 pm

Perhaps Jeremy (who is not on Github?) has some opinion/feedback on the big red button.

**Comment by ajordens** [comment]
Not against putting it on the config ... it's somewhat hidden but most people know/could easily be trained to look there.

**Comment by PoorImpulseCtrl** [comment]
I have some concerns about hiding it in config if we really want it to be a big red button. Maybe somewhere in the application sub-nav bar, just a red icon with some hover text/context.

Alternatively, having an actual box on the config page for ownership information (we want this for other reasons too), pulled from WP/krieger, with a link *there* for paging.

**Comment by anotherchrisberry** [comment]
I like it in the application header:

screen shot 2016-06-22 at 9 19 18 am

**Comment by anotherchrisberry** [comment]
(HTML for prior comment):
```html
  <li style="padding:10px 0 0 10px">
    <button class="btn btn-xs btn-danger" 
            ng-click="ctrl.pageApplicationOwner()" 
            ng-if="application.attributes.pdApiKey"
            uib-tooltip="Page application owner">
      <span class="glyphicon glyphicon-phone-alt"></span>
    </button>
  </li>
```

File: app/scripts/modules/core/application/application.controller.js [hunk 2]
```diff
@@ -68,8 +69,16 @@ module.exports = angular.module('spinnaker.application.controller', [
     ];
 
     applicationHotkeys.forEach(hotkeyBind.add);
-
     app.enableAutoRefresh($scope);
+
+    this.pageApplicationOwner = function pageApplicationOwner() {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/application/application.controller.js]:
```javascript
this.pageApplicationOwner = () => {
```
is preferred

File: app/scripts/modules/core/application/modal/pageApplicationOwner.modal.controller.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      var submitMethod = function () {
+        return applicationWriter.pageApplicationOwner(
+          $scope.application, $scope.command.reason
+        );
+      };
+
+      $scope.taskMonitor = taskMonitorService.buildTaskMonitor(taskMonitorConfig);
+      $scope.taskMonitor.submit(submitMethod);
+    };
+  });
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/application/modal/pageApplicationOwner.modal.controller.js]: Avoid using `$scope` on new code - we are trying to keep everything on the controller, so just change a few things:
```javascript
     this.application = application;
     this.command = {};
 
     this.submit = () => {
       var taskMonitorConfig = {
         application: application,
         title: 'Paging ' + application.name + ' owner',
         modalInstance: $uibModalInstance
       };
 
       var submitMethod = () => {
         return applicationWriter.pageApplicationOwner(application, this.command.reason);
       };
 
       this.taskMonitor = taskMonitorService.buildTaskMonitor(taskMonitorConfig);
       this.taskMonitor.submit(submitMethod);
     };
```
You'll need to tweak those references in the template, too - just change `application` to `ctrl.application`, `command` to `ctrl.command`, `taskMonitor` to `ctrl.taskMonitor`, etc.

Comment (ajordens) [inline app/scripts/modules/core/application/modal/pageApplicationOwner.modal.controller.js]: Roger.

**Pull request merged**